cmd/relayer/main.go

```diff
@@ -0,0 +1,412 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/urfave/cli/v2"
+
+	p2pnet "github.com/athanorlabs/go-p2p-net"
+	rcommon "github.com/athanorlabs/go-relayer/common"
+	rcontracts "github.com/athanorlabs/go-relayer/impls/gsnforwarder"
+	net "github.com/athanorlabs/go-relayer/net"
+	"github.com/athanorlabs/go-relayer/relayer"
+	rrpc "github.com/athanorlabs/go-relayer/rpc"
+
+	"github.com/athanorlabs/atomic-swap/cliutil"
+	"github.com/athanorlabs/atomic-swap/common"
+	contracts "github.com/athanorlabs/atomic-swap/ethereum"
+	swapnet "github.com/athanorlabs/atomic-swap/net"
+
+	logging "github.com/ipfs/go-log"
+)
+
+const (
+	flagDataDir          = "datadir"
+	flagEndpoint         = "endpoint"
+	flagForwarderAddress = "forwarder-address"
+	flagKey              = "key"
+	flagRPC              = "rpc"
+	flagRPCPort          = "rpc-port"
+	flagDeploy           = "deploy"
+	flagLog              = "log-level"
+	// TODO: do we need this, or can we assume all swap relayers will need to be on the p2p network?
+	flagWithNetwork       = "with-network"
+	flagLibp2pKey         = "libp2p-key"
+	flagLibp2pPort        = "libp2p-port"
+	flagBootnodes         = "bootnodes"
+	flagRelayerCommission = "relayer-commission"
+
+	defaultLibp2pPort = 10900
+)
+
+var (
+	log = logging.Logger("main")
+
+	flags = []cli.Flag{
+		&cli.StringFlag{
+			Name:  flagDataDir,
+			Usage: "Path to store swap artifacts",
+			Value: "{HOME}/.atomicswap/{ENV}", // For --help only, actual default replaces variables
+		},
+		&cli.StringFlag{
+			Name:  flagEndpoint,
+			Value: "http://localhost:8545",
+			Usage: "Ethereum RPC endpoint",
+		},
+		&cli.StringFlag{
+			Name:  flagKey,
+			Value: "eth.key",
```

Comment:
Do we want this to be in the current working directory?  This is what we use with `swapd`: https://github.com/AthanorLabs/atomic-swap/blob/master/cmd/swapd/main.go#L104-L108

Reply:
I was thinking we might integrate the relayer with swapd, so we could have transaction locking on the wallet. If the relayer is a separate daemon, we'll need to make sure the end user knows that they need to use separate wallet keys for the relayer and any swapd instances.

Reply:
updated